lxd/images.go

```diff
@@ -2886,7 +2878,10 @@ func imageDelete(d *Daemon, r *http.Request) response.Response {
 		}
 
 		// Remove main image file from disk.
-		imageDeleteFromDisk(details.image.Fingerprint)
+		err = imageDeleteFromDisk(details.image.Fingerprint)
```

Comment:
We should probably remove the DB record above last otherwise if the image delete fails the image record has now been removed so we can retry it.

Reply:
Thanks, I've moved the call to `imageDeleteFromDisk` before the DB record removal